crypto/bls.go

```diff
@@ -82,8 +70,8 @@ func (sk *PrKeyBLSBLS12381) Sign(data []byte, kmac hash.Hasher) (Signature, erro
 	// hash the input to 128 bytes
 	h := kmac.ComputeHash(data)
 
-	// intialize BLS context
-	_ = newBLSBLS12381()
+	// set BLS context
+	blsInstance.reInit()
```

Comment:
> If the implementation evolves and relic has multiple contexts,
> reinit should be called at every a. operation.

So this is kind of a safety measure for expected changes in the relic implementation? Or relic configuration?

Reply:
Relic library has settings stored in a context structure. It only supports one context at a time.
If we use Relic for BLS on the specific curve we're using (BLS 12-381), we need to generate and set that context (happens at `init()`). 
We're currently using Relic with only one context, so we could just `init()` once and forget about it. But if we were to add more algorithms or curves, we will have to set the right context before every use. `reInit()` sets that context that is previously generated and stored. `reInit()` is just setting one pointer so it's inexpensive and makes sure there are no problems if we use Relic for other things.